src/librustc_mir/borrow_check/move_errors.rs

```diff
@@ -237,17 +259,11 @@ impl<'a, 'gcx, 'tcx> MirBorrowckCtxt<'a, 'gcx, 'tcx> {
                                 .tcx
                                 .cannot_move_out_of_interior_noncopy(span, ty, None, origin),
                             ty::TyClosure(def_id, closure_substs)
-                                if !self.mir.upvar_decls.is_empty()
-                                    && {
-                                        match place {
-                                            Place::Projection(ref proj) => {
-                                                proj.base == Place::Local(Local::new(1))
-                                            }
-                                            Place::Promoted(_) |
-                                            Place::Local(_) | Place::Static(_) => unreachable!(),
-                                        }
-                                    } =>
-                            {
+                                if !self.mir.upvar_decls.is_empty() &&
+                                    original_path.strip_deref_projections()
```

Comment:
hmm, is `strip_deref_projections` enough?

Try a case like this:

```rust
{
    let x = (vec![22], vec![44]);
    expect_fn(|| drop(x.0));
}
```

does that work?

Reply:
Fixed.